### PR TITLE
Refactor the keypair management of sequencer, proposer, and relayer

### PR DESCRIPTION
**Pull request by baichuan3**
resolve https://github.com/rooch-network/rooch/issues/800

Summary
1. Init server config when `Rooch init`, and generate rooch key pair in keystore, both for sequencer, proposer, and relayer

Notice
Due to dependencies (Keystore is in WalletContext), the server loads key pairs from the command line by default and then passes them to the RPC Server. KeyPair does not have a clone attribute. The ServerOpt attribute is newly added to encapsulate and extend KeyPairs and other configurations.
There will be an impact on the SDK.

Todo
Wait the password option to decrypt private key for commands

### CODE REVIEW

File: crates/rooch-rpc-server/Cargo.toml [hunk 1]
```diff
@@ -37,7 +37,8 @@ tracing-subscriber = { workspace = true }
 schemars = { workspace = true }
 serde_with = { workspace = true }
 rand = { workspace = true }
-fastcrypto = { workspace = true }
+#fastcrypto = { workspace = true }
```
Comment (feliciss) [inline crates/rooch-rpc-server/Cargo.toml:40]: Why did not delete this line?

**Review by feliciss — COMMENTED** [review]
Didn't know the detailed design, but the code LGTM and the backend works.

File: crates/rooch-rpc-server/Cargo.toml (hunk 1 above)
Comment (baichuan3) [reply crates/rooch-rpc-server/Cargo.toml]: deleted

File: crates/rooch-types/Cargo.toml [hunk 2]
```diff
@@ -31,7 +31,8 @@ serde_json = { workspace = true }
 serde_with = { workspace = true }
 derive_more = { workspace = true }
 eyre = { workspace = true }
-fastcrypto = { workspace = true }
+#fastcrypto = { workspace = true }
```
[the remark this answers is not in the document]
Comment (baichuan3) [reply crates/rooch-types/Cargo.toml]: deleted

File: crates/rooch-rpc-server/src/lib.rs [hunk 3]
```diff
@@ -175,25 +176,46 @@ pub async fn run_start_server(opt: &RoochOpt) -> Result<ServerHandle> {
...
+    if server_opt.key_keypairs.is_empty() {
+        // only for test, generate test key pairs
+        if chain_id_opt.is_test_or_dev_or_local() {
+            let (_, key_keypair, _, _) = generate_new_key_pair::<RoochAddress, RoochKeyPair>(
+                KeyPairType::RoochKeyPairType,
+                None,
+                None,
+            )?;
+            server_opt.key_keypairs =
+                vec![key_keypair.copy(), key_keypair.copy(), key_keypair.copy()];
```
[the remark this answers is not in the document]
Comment (baichuan3) [reply crates/rooch-rpc-server/src/lib.rs]: resolved

**Comment by baichuan3** [comment]
The design is very simple. Use a extra key address as the keypair of sequencer, proposer and relayer. When rooch init, write the key address to the Server Config configuration. When the server starts, load the KeyPair of the Key Address from the configuration. @feliciss

File: crates/rooch/src/commands/init.rs [hunk 4]
```diff
@@ -16,29 +19,86 @@ use rooch_types::crypto::RoochKeyPair;
...
 
 /// Tool for init with rooch
 #[derive(Parser)]
 pub struct Init {
     /// Command line input of custom server URL
     #[clap(short = 's', long = "server-url")]
     pub server_url: Option<String>,
+    /// The address of the Rooch account to be set as key address
+    #[clap(short = 'k', long = "key-address")]
+    key_address: Option<String>,
```
Comment (jolestar) [inline crates/rooch/src/commands/init.rs:32]: I think we do not need the `key_address` when `init`, the `key_address` is just a server start option.

File: crates/rooch/src/commands/server/commands/start.rs [hunk 5]
```diff
@@ -25,16 +29,42 @@ pub struct StartCommand {
...
+            .client_config
+            .keystore
+            .get_key_pair_by_key_pair_type(&key_address, KeyPairType::RoochKeyPairType)
+            .map_err(|e| RoochError::KeyAddressKeyPairDoesNotExistError(e.to_string()))?;
+
+        // Construct sequencer, proposer and relayer keypair
+        let mut server_opt = ServerOpt::new();
+        server_opt.sequencer_keypair = Some(key_keypair.copy());
+        server_opt.proposer_keypair = Some(key_keypair.copy());
+        server_opt.relayer_keypair = Some(key_keypair.copy());
```
Comment (jolestar) [inline crates/rooch/src/commands/server/commands/start.rs:57]: If the node operator wants to configure a different address for sequencer|proposer|relayer,  it's impossible in the current solution.

How about we add `sequencer_account: Option<address>`, `proposer_account: Option<address>`, `relayer_account` to the `RoochOpt`?

If the option is None, we just use the active account. And we do not need the `key_address` config.

**Comment by feliciss** [comment]
> The design is very simple. Use a extra key address as the keypair of sequencer, proposer and relayer. When rooch init, write the key address to the Server Config configuration. When the server starts, load the KeyPair of the Key Address from the configuration. @feliciss

As mentioned by @jolestar, I do not think there's a need to add an extra key pair for server options. When the server starts, the user can choose the key pair in the store or possibly init it as server config.

File: crates/rooch/src/commands/server/commands/start.rs (hunk 5 above)
Comment (baichuan3) [reply crates/rooch/src/commands/server/commands/start.rs]: > If the node operator wants to configure a different address for sequencer|proposer|relayer, it's impossible in the current solution.
> 
> How about we add `sequencer_account: Option<address>`, `proposer_account: Option<address>`, `relayer_account` to the `RoochOpt`?
> 
> If the option is None, we just use the active account. And we do not need the `key_address` config.

Has remove key address from server config, and add `sequencer_account: Option<address>, proposer_account: Option<address>, relayer_account` to the RoochOpt

**Comment by baichuan3** [comment]
Find a bug about `account account create` and ``account account list`, the newly created account will be overwritten in the keystore and cannot be displayed in the account list. It will be located and resolved in a separate PR.

**Review by feliciss — APPROVED** [review]
It needs to handle address lengthy check or reload the address after detecting modification of the addresses in `rooch.yaml` file:

```
keystore: !File
  keystore:
    keys:
      0xde055ca33ed7ad22830e32bb1a75cff79c69c929cfe1159268434f18467bbb62:
        RoochKeyPairType: AARyVhWm2ufvLRSqT0Wz4wnsMxzTww1uKkga1Ud91SAj
    session_keys: {}
  path: /Users/system/.rooch/rooch_config/rooch.keystore
active_address: 0xde055ca33ed7ad22830e32bb1a75cff79c69c929cfe1159268434f18467bbb62
```

After change:

```
keystore: !File
  keystore:
    keys:
      0xde055ca33ed7ad22830e32bb1a75cff79c69c929cfe1159268434f18467bbb:
        RoochKeyPairType: AARyVhWm2ufvLRSqT0Wz4wnsMxzTww1uKkga1Ud91SAj
    session_keys: {}
  path: /Users/system/.rooch/rooch_config/rooch.keystore
active_address: 0xde055ca33ed7ad22830e32bb1a75cff79c69c929cfe1159268434f18467bbb
```

Server output for both:

```
2023-10-03T16:59:43.375180Z  INFO moveos_common::utils: set max open fds 45056    
2023-10-03T16:59:43.375300Z  INFO rooch_rpc_server: Load latest state root 0xe568cf23ad4a2f0bd08e98fdee609f6681d721d0f6ee067f55f56108e4d9beef
2023-10-03T16:59:43.794252Z  INFO rooch_rpc_server: RPC Server sequencer address: 0xde055ca33ed7ad22830e32bb1a75cff79c69c929cfe1159268434f18467bbb62
2023-10-03T16:59:43.794290Z  INFO rooch_sequencer::actor::sequencer: Load latest sequencer order 0
2023-10-03T16:59:43.794362Z  INFO rooch_rpc_server: RPC Server proposer address: 0xde055ca33ed7ad22830e32bb1a75cff79c69c929cfe1159268434f18467bbb62
```

**Comment by baichuan3** [comment]
I will merge this PR first, and handle the output in another PR

**Pull request merged**